src/main/java/org/broadinstitute/dsm/model/ups/UPSActivity.java

```diff
@@ -8,4 +12,43 @@
     UPSStatus status;
     String date;
     String time;
+
+    public UPSActivity() {}
+
+    public UPSActivity(UPSStatus status, String date, String time) {
+        this.status = status;
+        this.date = date;
+        this.time = time;
+    }
+
+    public String getDateTimeString() {
+        return date + " " + time;
+    }
+
+    public Instant getInstant() {
+        Instant eventTime = null;
+        String dateTime = getDateTimeString();
+        if (dateTime != null) {
+            try {
+                eventTime = new SimpleDateFormat("yyyyMMdd kkmmss").parse(dateTime).toInstant();
```

Comment:
1) What timezone are the UPS timestamps in? Are they in UTC? I believe `SimpleDateFormat` assumes _local_ timezone. Is DSM always running in UTC timezone or EST timezone? You might want to call `setTimeZone()` to explicitly set UTC timezone.

2) I recommend using the newer class `DateTimeFormatter`.

3) Curious why the hour pattern is `kk` instead of `HH`? Is that a peculiarity of the UPS timestamps?

Reply:
According to the UPS docs, the time zone is local time for the place where the event happened.  So figuring this out would need a bit more work--looking at the `location` object and figuring out the zone.  From the docs, I can't see an obvious place to get the zone from the API.

Hour should be `HH`!  Thanks for that.

Reply:
@yufengwng one more round?  Given the time sensitive (hahaha pun) nature of this and the geographic constraints of testboston, I went for `America/New_York` and documented things accordingy.